lib/rspec/core/example_group.rb

```diff
@@ -769,6 +769,11 @@ def initialize(shared_group_name, inclusion_location)
         @inclusion_location = inclusion_location
       end
 
+      # @private
+      def ==(other)
+        @shared_group_name == other.shared_group_name
+      end
```

Comment:
This is just a short cut so we can check if theres an existing group with that name in our current call stack, whilst we do allow groups to share the same name, they will never be in the same call stack.